relay-metrics/src/protocol.rs

```diff
@@ -349,20 +419,27 @@ fn parse_tags(string: &str) -> Option<BTreeMap<String, String>> {
 
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 pub struct Metric {
-    /// The name of the metric without its unit.
+    /// The name of the metric, i.e. the MRI.
     ///
-    /// Metric names cannot be empty, must start with a letter and can consist of ASCII
-    /// alphanumerics, underscores, slashes, @s and periods.
-    pub name: String,
-    /// The unit of the metric value.
+    /// The metric name historically has been a freeform string. Over time we realized that we can
+    /// use the metric name to encode things like metric unit, type, etc in order to force
+    /// downstream systems to bucket by those properties, without having to update them. This is
+    /// how the concept of the Metrics Resource Identifier was born.
     ///
-    /// Units augment metric values by giving them a magnitude and semantics. There are certain
-    /// types of units that are subdivided in their precision, such as the [`DurationUnit`] for time
-    /// measurements.
+    /// An MRI has the following format: `<type>:<namespace>/<name>@<unit>`
     ///
-    /// The unit can be omitted and defaults to [`MetricUnit::None`].
-    #[serde(default, skip_serializing_if = "MetricUnit::is_none")]
-    pub unit: MetricUnit,
+    ///
+    /// * `type` is the single-letter representation of the metric's type, e.g. one of `d`, `s`, `c`.
+    /// * `namespace` is one of the values representable by `MetricNamespace`.
+    /// * `name` cannot be empty, must start with a letter and can consist of ASCII alphanumerics, underscores, slashes, @s and periods.
+    /// * `unit` is one of the values representable by `MetricUnit`.
+    ///
+    /// For parsing and normalization, the [`MetricResourceIdentifier`] struct is used in the aggregator. It is
+    /// also used in the kafka producer to route certain namespaces to certain topics.
+    ///
+    /// Note that the format used in the statsd protocol is different: Metric names are not prefixed
+    /// with `<ty>:` as the type is somewhere else in the protocol.
```

Comment:
This note should probably be moved to the crate-level documentation of the statsd protocol.